### PR TITLE
fix: confirm modal firing onOk event twice

### LINKED ISSUE #40272: Modal confirm resolved function can be fired twice
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/confirmation-modal-dialog-antd-5-1-5-forked-9nmp96?file=/demo.tsx)

### Steps to reproduce

Click on the "with promise" button, then click on the "Ok" button and click it several times really fast while it still loading. 
If you are fast enough you can see that the button can be clicked again an the function fired again before the modal finishes closing.

### What is expected?

The button to not fire the function while it is loading and the modal is closing.

### What is actually happening?

If you are fast enough you can click the button while the modal is closing and it ends up firing the function again.

https://user-images.githubusercontent.com/87539237/212683977-aae26e63-ea55-4634-bd6d-d9550100707b.mp4

| Environment | Info |
| --- | --- |
| antd | 5.1.5 |
| React | ^18.2.0 |
| System | Windows 10 |
| Browser | Brave |

---
This problem can be more easily reproduced on lower end systems that have a bigger window on unmounting components on the screen, so its easier to click the button while the modal is closing.

### PR DESCRIPTION
**Pull request by Rafael-Martins**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #40272 

### 💡 Background and solution
This is my first contribution guys :smile:, not sure if I'm missing something!
 
We have a render when the confirm button using promise finished loading and can be clickable again, letting the user trigger onOk function for a second time 
The solution was set a disabled state when the button is loading, if the promise resolves we close de modal with the button disabled, if the promise rejects we remove the disabled from the button.

### 📝 Changelog
Changed the onOk button behavior to be disabled when a promise is pending

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix: confirm modal firing onOk event twice        |
| 🇨🇳 Chinese |   修复: confirm modal onOk 可能触发两次的问题        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/_util/ActionButton.tsx [hunk 1]
```diff
@@ -111,6 +114,7 @@ const ActionButton: React.FC<ActionButtonProps> = (props) => {
       {...convertLegacyProps(type)}
       onClick={onClick}
       loading={loading}
+      disabled={waitingPromise}
```
Comment (afc163) [inline components/_util/ActionButton.tsx:117]: I prefer to keeping loading style after press button.

**Comment by Rafael-Martins** [comment]
Hey @afc163 sorry bother you but, what do you think about it? just asking to avoid this get stale ;D

**Review by yoyo837 — APPROVED** [review]

**Comment by hxing** [comment]
The change brings one issue. If I trigger the onOK twice. The waiting icon is there.

**Comment by yoyo837** [comment]
> 

Could you provide a reproduction and open a new issue for that?

**Comment by hxing** [comment]
https://ant.design/components/popconfirm-cn

> > 
> 
> Could you provide a reproduction and open a new issue for that?

The demo of "Popcon confirm with promise" in https://ant.design/components/popconfirm-cn
Please try to click the button again after the confirm disappears. antd 5.2.2 is ok

**Comment by zombieJ** [comment]
> The demo of "Popcon confirm with promise" in https://ant.design/components/popconfirm-cn Please try to click the button again after the confirm disappears. antd 5.2.2 has no such issue

Seems this issue is large than the fix. Let me check better way.

**Pull request merged**